eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ExternalDependencyExposedCheck.java

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FullIdent;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ *  No external dependency exposed in the public API
+ */
+public class ExternalDependencyExposedCheck extends AbstractCheck {
+    private static final String JAVA = "java";
+    private static final String COM_AZURE = "com.azure";
+    private static final String REACTOR = "reactor";
+    private static final String DOT = ".";
+
+    private static final String EXTERNAL_DEPENDENCY_ERROR =
+        "Class ''%s'', is a class from external dependency. You should not use it as a return or parameter argument type";
+
+    private static final Set<String> validLibrarySet = new HashSet<>();
```

Comment:
Why static final? means there can only be one instance of it.

Reply:
`validDependencyPackageNames` or something to that effect is a better name.

Reply:
want to have a fixed set of libraries to only contain the non-external libraries.  I made the set to be a fixed set because the set will not change.  
```
    private static final Set<String> VALID_DEPENDENCY_PACKAGE_NAMES = Collections.unmodifiableSet(new HashSet<String>(Arrays.asList(
        JAVA, COM_AZURE, REACTOR
    )));
```